packages/astro-org/src/index.ts

```diff
@@ -51,87 +67,79 @@ export default function org(options: OrgPluginOptions = {}): AstroIntegration {
-              // Astro typing requires slug to be a string, however
-              // I'm pretty sure that mdx integration returns
-              // undefined if slug is not set in frontmatter.
-              slug: frontmatter.slug as any,
+              // NOTE: Astro typing requires slug to be a string, however I'm
+              // pretty sure that mdx integration returns undefined if slug is
+              // not set in frontmatter.
+              slug: frontmatter.slug as string,
               rawData: contents,
             };
           },
-          contentModuleTypes: await fs.readFile(
+          async getRenderModule({ fileUrl, contents }) {
+            const pluginContext = this;
+            const filePath = fileURLToPath(fileUrl);
+
+            const f = new VFile({ path: fileUrl, value: contents });
+            const hast = await uniorgToHast.run(
+              uniorgToHast.parse(f) as OrgData,
+              f
+            );
+
+            await emitOptimizedImages(hast, {
+              astroConfig,
+              pluginContext,
+              filePath,
+            });
+
+            // TODO(Kevin): Typescript mismatch about the same packag?
+            const htmlStr = htmlToHtml.stringify(hast as any, f);
+
+            const code = `
+import { jsx, Fragment } from 'astro/jsx-runtime';
+const html = ${JSON.stringify(htmlStr)}
+
+export async function Content(props) {
+    return jsx(Fragment, { 'set:html': html });
+}
+export default Content;
+`;
+            return { code };
+          },
+
+          contentModuleTypes: fs.readFileSync(
             new URL('../template/content-module-types.d.ts', import.meta.url),
             'utf-8'
           ),
           handlePropagation: true,
         });
-        updateConfig({
-          vite: {
-            plugins: [
-              {
-                enforce: 'pre',
-                configResolved(resolved: any) {
-                  // HACK: move ourselves before Astro's JSX plugin to transform things in the right order
-                  const jsxPluginIndex = resolved.plugins.findIndex(
-                    (p: any) => p.name === 'astro:jsx'
-                  );
-                  if (jsxPluginIndex !== -1) {
-                    const myPluginIndex = resolved.plugins.findIndex(
-                      (p: any) => p.name === 'rollup-plugin-orgx'
-                    );
-                    if (myPluginIndex !== -1) {
-                      const myPlugin = resolved.plugins[myPluginIndex];
-                      // @ts-ignore-error ignore readonly annotation
-                      resolved.plugins.splice(myPluginIndex, 1);
-                      // @ts-ignore-error ignore readonly annotation
-                      resolved.plugins.splice(jsxPluginIndex, 0, myPlugin);
-                    }
-                  }
-                },
-                ...orgPlugin({
-                  ...options,
-                  uniorgPlugins,
-                  rehypePlugins: [
```

Comment:
To be honest, I'm not sure what this mess was doing. I completely escaped `orgx` as I didn't understand the magic it's doing.